src/main/java/io/cryostat/core/EventOptionsBuilder.java

```diff
@@ -0,0 +1,147 @@
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.core;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.openjdk.jmc.common.unit.IConstrainedMap;
+import org.openjdk.jmc.common.unit.IConstraint;
+import org.openjdk.jmc.common.unit.IMutableConstrainedMap;
+import org.openjdk.jmc.common.unit.IOptionDescriptor;
+import org.openjdk.jmc.common.unit.QuantityConversionException;
+import org.openjdk.jmc.flightrecorder.configuration.events.EventOptionID;
+import org.openjdk.jmc.flightrecorder.configuration.events.IEventTypeID;
+import org.openjdk.jmc.rjmx.IConnectionHandle;
+import org.openjdk.jmc.rjmx.ServiceNotAvailableException;
+import org.openjdk.jmc.rjmx.services.jfr.IEventTypeInfo;
+import org.openjdk.jmc.rjmx.services.jfr.internal.FlightRecorderServiceV2;
+
+import io.cryostat.core.net.JFRConnection;
+import io.cryostat.core.tui.ClientWriter;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public class EventOptionsBuilder {
+
+    private final boolean isV2;
+    private final IMutableConstrainedMap<EventOptionID> map;
+    private Map<IEventTypeID, Map<String, IOptionDescriptor<?>>> knownTypes;
+    private Map<String, IEventTypeID> eventIds;
+
+    public EventOptionsBuilder(ClientWriter cw, JFRConnection connection, Supplier<Boolean> v2)
+            throws ServiceNotAvailableException, IOException,
+                    org.openjdk.jmc.rjmx.services.jfr.FlightRecorderException {
+        this.isV2 = v2.get();
+        this.map = connection.getService().getDefaultEventOptions().emptyWithSameConstraints();
+        knownTypes = new HashMap<>();
+        eventIds = new HashMap<>();
+
+        if (!isV2) {
+            cw.println("Flight Recorder V1 is not supported");
```

Comment:
Should this be `Flight Recorder V2 is not supported`?

Reply:
I think it's right - only JFR V2 is supported, not V1 (which is the old pre-OpenJDK opensourced implementation), so this message is emitted if the connection is not V2.

Reply:
Ahhh okayy I got a little confused there^^Thanks!